engines/dfc_provider/app/services/dfc_catalog.rb

```diff
@@ -53,4 +54,27 @@ def adjust_to_wholesale_price(broker, product)
     offer.price = wholesale_variant_price.dup
     offer.price.value = offer.price.value.to_f / transformation.factor
   end
+
+  def adjust_to_wholesale_stock(product, transformation)
+    adjust_item_stock(product, transformation)
+    adjust_offer_stock(product, transformation)
+  end
+
+  def adjust_item_stock(product, transformation)
+    item = product.catalogItems&.first
+    wholesale_item = transformation.product.catalogItems&.first
+
+    return unless item && wholesale_item&.stockLimitation.present?
+
+    item.stockLimitation = wholesale_item.stockLimitation.to_i * transformation.factor
+  end
+
+  def adjust_offer_stock(product, transformation)
+    offer = product.catalogItems&.first&.offers&.first
+    wholesale_offer = transformation.offer
+
+    return unless offer && wholesale_offer&.stockLimitation.present?
+
+    offer.stockLimiation = wholesale_offer.stockLimitation.to_i * transformation.factor
```

Comment:
Hmm, that spelling doesn't look right:

```suggestion
    offer.stockLimitation = wholesale_offer.stockLimitation.to_i * transformation.factor
```

The test below isn't testing for this. Is it something that can be tested?

Reply:
Well spotted!

While the standard allows a stock limitation on catalog items and offers, the FDC data stores the stock only on the catalog item and not the offer. So this code path didn't run. I added a test case for that now.